### ISSUE TITLE
Support partial chunk management

### ISSUE THREAD

**Issue by zhuzilin** [Tencent/PatrickStar#188]
As we hope to support MoE in #187, and MoE is mainly of model parallel structure instead of data parallel, we need to support managing only part of the model with chunk.

There are several design choices need to make, including but not limited to:
- Shall we use mixed precision training in the unmanaged part of the model
- How could we connect the backward of the unmanaged parts with the managed parts, i.e. if there are 3 parts in the model:
  ```python
  class Net(nn.Module):
    def __init__(self, ...):
      self.A = SubNetA(...)  # managed by chunk
      self.B = SubNetB(...)  # not managed by chunk
      self.C = SubNetC(...)  # managed by chunk
  ```
  Then `self.A` and `self.C` need `model.backward(loss)` while `self.B` only need `loss.backward()`.

cc @feifeibear

**Comment by feifeibear** [comment]
1. what do you mean by precision training?
2. I think you mean we can define a hybrid computation graph. 
A part of the graph is managed by chunk-based mem mgr, and the rest of the graph is managed by Pytorch. Is it similar to the CPU embedding optimization?
Therefore the autograd provided by PyTorch is not affected in this way?

**Comment by zhuzilin** [comment]
It seems that we could add a scope, inside which all parameter created will be `TORCH_BASED`.

**Comment by zhuzilin** [comment]
@feifeibear
> what do you mean by precision training?

mixed precision training. train with fp16.

> I think you mean we can define a hybrid computation graph.
A part of the graph is managed by chunk-based mem mgr, and the rest of the graph is managed by Pytorch. Is it similar to the CPU embedding optimization?
Therefore the autograd provided by PyTorch is not affected in this way?

I agree. We could make the not managed part the same as the CPU embedding opt.

**Comment by feifeibear** [comment]
> @feifeibear
> 
> > what do you mean by precision training?
> 
> mixed precision training. train with fp16.
> 
> > I think you mean we can define a hybrid computation graph.
> > A part of the graph is managed by chunk-based mem mgr, and the rest of the graph is managed by Pytorch. Is it similar to the CPU embedding optimization?
> > Therefore the autograd provided by PyTorch is not affected in this way?
> 
> I agree. We could make the not managed part the same as the CPU embedding opt.
